packages/shared/src/hooks/post/useBlockPost.ts

```diff
@@ -0,0 +1,239 @@
+import { useQuery, useQueryClient } from 'react-query';
+import { useCallback, useMemo } from 'react';
+import { Post } from '../../graphql/posts';
+import { generateStorageKey, StorageTopic } from '../../lib/storage';
+import { useAuthContext } from '../../contexts/AuthContext';
+import { useActions } from '../useActions';
+import { ActionType } from '../../graphql/actions';
+import useTagAndSource from '../useTagAndSource';
+import { Origin } from '../../lib/analytics';
+import {
+  BlockTagSelection,
+  DownvoteBlocked,
+  getBlockedLength,
+  getBlockedMessage,
+} from '../../components/post/block/common';
+import { useLazyModal } from '../useLazyModal';
+import { LazyModal } from '../../components/modals/common/types';
+import { disabledRefetch, isNullOrUndefined } from '../../lib/func';
+import { useToastNotification } from '../useToastNotification';
+
+interface BlockData {
+  showTagsPanel?: boolean;
+  blocked?: DownvoteBlocked;
+}
+
+interface UseBlockPost {
+  data: BlockData;
+  blockedTags: number;
+  onClose(forceClose?: boolean): void;
+  onShowPanel(): void;
+  onDismissPermanently(): void;
+  onReport(): void;
+  onUndo(): void;
+  onBlock(tags: BlockTagSelection, shouldBlockSource: boolean): void;
+}
+
+interface UseBlockPostProps {
+  toastOnSuccess?: boolean;
+  blockedSource?: boolean;
+}
+
+interface Params {
+  blocks: string[];
+  unblocks: string[];
+}
+
+const getParams = (tags: BlockTagSelection): Params => {
+  const blocks = [];
+  const unblocks = [];
+
+  if (tags) {
+    Object.entries(tags).forEach(([tag, shouldBlock]) => {
+      const container = shouldBlock ? blocks : unblocks;
+      container.push(tag);
+    });
+  }
+
+  return { blocks, unblocks };
+};
+
+const ignoredCall = () => Promise.resolve({ successful: true });
+
+export const useBlockPost = (
```

Comment:
From review point of view as someone looking at first time.
- Very big file, lots of things going on, not super clear what's happening really.

Ideally it would be either better named, or split. (non blocking comment)